pkg/gardener/shoot/hyperscaler/aws/config.go

```diff
@@ -44,13 +44,19 @@ func DecodeInfrastructureConfig(data []byte) (*v1alpha1.InfrastructureConfig, er
 }
 
 func NewInfrastructureConfig(workersCidr string, zones []string) v1alpha1.InfrastructureConfig {
+	awsZones, err := generateAWSZones(workersCidr, zones)
+	if err != nil {
+		// TODO - pass the error to caller?
```

Comment:
Please do not forget that the error should be handled

Reply:
Could you consider:
1) reflecting that in runtimeCR status
2) whether logging the information about the error would make sense

Reply:
The error will be handled as standard conversion error during shoot create/update with wrapped appropriate error message in the status. The code: 

```golang
	shoot, err := convertCreate(&s.instance, gardener_shoot.CreateOpts{
		ConverterConfig:       m.ConverterConfig,
		AuditLogData:          data,
		MaintenanceTimeWindow: getMaintenanceTimeWindow(s, m),
		StructuredAuthEnabled: m.StructuredAuthEnabled,
	})
	if err != nil {
		m.log.Error(err, "Failed to convert Runtime instance to shoot object")
		m.Metrics.IncRuntimeFSMStopCounter()
		return updateStatePendingWithErrorAndStop(
			&s.instance,
			imv1.ConditionTypeRuntimeProvisioned,
			imv1.ConditionReasonConversionError,
			"Runtime conversion error")
	} 
```